memory.py

```diff
@@ -15,46 +14,71 @@ def __init__(self, size):
     self.index = 0
     self.size = size
     self.full = False  # Used to track actual capacity
-    self.sum_tree = np.zeros((2 * size - 1, ), dtype=np.float32)  # Initialise fixed size tree with all (priority) zeros
-    self.data = np.array([None] * size)  # Wrap-around cyclic buffer
+    self.tree_start = 1
+    while self.tree_start < self.size: # Put all used node leaves on last tree level
```

Comment:
If this is done on init, the formula could be worked out and used instead of using a `while` statement?

Reply:
I have replaced the while loop with an exact formula.